k4MarlinWrapper/src/components/Lcio2EDM4hep.cpp

```diff
@@ -131,9 +146,12 @@ namespace {
 
 StatusCode Lcio2EDM4hepTool::convertCollections(lcio::LCEventImpl* the_event) {
   // Convert event parameters
-  // auto& frame = const_cast<podio::Frame&>(m_podioDataSvc->getEventFrame());
-  auto& frame = m_podioDataSvc->m_eventframe;
-  LCIO2EDM4hepConv::convertObjectParameters<lcio::LCEventImpl>(the_event, frame);
+  if (m_podioDataSvc) {
+    LCIO2EDM4hepConv::convertObjectParameters<lcio::LCEventImpl>(the_event, m_podioDataSvc->m_eventframe);
+  } else {
+    LCIO2EDM4hepConv::convertObjectParameters<lcio::LCEventImpl>(
```

Comment:
```suggestion
    LCIO2EDM4hepConv::convertObjectParameters(the_event, m_podioDataSvc->m_eventframe);
  } else {
    LCIO2EDM4hepConv::convertObjectParameters(
```
We don't have to specify this as it can be deduced.

Reply:
Fixed